probe-rs/src/probe/daplink/mod.rs

```diff
@@ -213,6 +230,112 @@ impl DAPLink {
             _ => Ok(0),
         }
     }
+
+    /// Set SWO port to use requested transport.
+    ///
+    /// Check the probe capabilities to determine which transports are available.
+    fn set_swo_transport(
+        &mut self,
+        transport: swo::TransportRequest,
+    ) -> Result<(), DebugProbeError> {
+        let response = commands::send_command(&mut self.device, transport)?;
+        match response {
+            swo::TransportResponse(Status::DAPOk) => Ok(()),
+            swo::TransportResponse(Status::DAPError) => Err(CmsisDapError::UnexpectedAnswer.into()),
+        }
+    }
+
+    /// Set SWO port to specified mode.
+    ///
+    /// Check the probe capabilities to determine which modes are available.
+    fn set_swo_mode(&mut self, mode: swo::ModeRequest) -> Result<(), DebugProbeError> {
+        let response = commands::send_command(&mut self.device, mode)?;
+        match response {
+            swo::ModeResponse(Status::DAPOk) => Ok(()),
+            swo::ModeResponse(Status::DAPError) => Err(CmsisDapError::UnexpectedAnswer.into()),
+        }
+    }
+
+    /// Set SWO port to specified baud rate.
+    ///
+    /// Returns `SWOBaudrateNotConfigured` if the probe returns 0,
+    /// indicating the requested baud rate was not configured,
+    /// and returns the configured baud rate on success (which
+    /// may differ from the requested baud rate).
+    fn set_swo_baudrate(&mut self, baud: swo::BaudrateRequest) -> Result<u32, DebugProbeError> {
+        let response: swo::BaudrateResponse = commands::send_command(&mut self.device, baud)?;
+        debug!("Requested baud {}, got {}", baud.0, response.0);
+        if response.0 == 0 {
+            Err(CmsisDapError::SWOBaudrateNotConfigured.into())
+        } else {
+            Ok(response.0)
+        }
+    }
+
+    /// Start SWO trace data capture.
+    fn start_swo_capture(&mut self) -> Result<(), DebugProbeError> {
+        let response = commands::send_command(&mut self.device, swo::ControlRequest::Start)?;
+        match response {
+            swo::ControlResponse(Status::DAPOk) => Ok(()),
+            swo::ControlResponse(Status::DAPError) => Err(CmsisDapError::UnexpectedAnswer.into()),
+        }
+    }
+
+    /// Stop SWO trace data capture.
+    fn stop_swo_capture(&mut self) -> Result<(), DebugProbeError> {
+        let response = commands::send_command(&mut self.device, swo::ControlRequest::Stop)?;
+        match response {
+            swo::ControlResponse(Status::DAPOk) => Ok(()),
+            swo::ControlResponse(Status::DAPError) => Err(CmsisDapError::UnexpectedAnswer.into()),
+        }
+    }
+
+    /// Fetch current SWO trace status.
+    #[allow(dead_code)]
+    fn get_swo_status(&mut self) -> Result<swo::StatusResponse, DebugProbeError> {
+        Ok(commands::send_command(
+            &mut self.device,
+            swo::StatusRequest {},
```

Comment:
```suggestion
            swo::StatusRequest,
```

I am not sure if this is really better. I was just confused for a second if the {} is really necessary :)